core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -230,12 +251,16 @@ class NewHadoopRDD[K, V](
     super.persist(storageLevel)
   }
 
-
-  def getConf: Configuration = confBroadcast.value.value
```

Comment:
I'm not sure how that helps. You'll want the Configuration to be cloned every time it's called in order to fix the concurrent modification issue. If you make it a lazy val, the lazily evaluated value will still be shared by multiple threads.